packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -454,11 +458,14 @@ class _SearchAnchorState extends State<SearchAnchor> {
   Widget build(BuildContext context) {
     return AnimatedOpacity(
       key: _anchorKey,
-      opacity: _anchorIsVisible ? 1.0 : 0.0,
+      opacity: widget.enabled ? _anchorIsVisible ? 1.0 : 0.0 : _kDisableSearchBarOpacity,
```

Comment:
This double ternary is hard to read. Could you use a `switch`?

Reply:
not sure of what you're loking for here. You want the result to be defined outside the widget?